tensorboard/webapp/widgets/histogram/histogram_v2_test.ts

```diff
@@ -0,0 +1,249 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Component, DebugElement, Input} from '@angular/core';
+import {ComponentFixture, TestBed} from '@angular/core/testing';
+import {By} from '@angular/platform-browser';
+import {NoopAnimationsModule} from '@angular/platform-browser/animations';
+
+import {
+  Bin,
+  ColorScale,
+  HistogramData,
+  HistogramDatum,
+  HistogramMode,
+  TimeProperty,
+} from './histogram_types';
+import {HistogramV2Component} from './histogram_v2_component';
+
+function buildBin(override: Partial<Bin> = {}): Bin {
+  return {
+    x: 0,
+    dx: 1,
+    y: 0,
+    ...override,
+  };
+}
+
+function buildHistogramDatum(
+  override: Partial<HistogramDatum> = {}
+): HistogramDatum {
+  return {
+    wallTime: 1000,
+    step: 0,
+    bins: new Array(10).fill(buildBin()),
```

Comment:
Presumably this function should return bins whose elements are not exactly the same object reference?
e.g. something like
`new Array(10).fill(0).map(x => buildBin())`